docs/root/api/starting_envoy.rst

```diff
@@ -10,29 +10,130 @@ Starting Envoy
-functions, some of which are demonstrated below:
+After the client is created, it should be stored and kept in memory in order to be used
+for issuing requests.
 
 **Kotlin example**::
 
-  val envoy = AndroidEnvoyClientBuilder(baseContext)
+  val envoy = AndroidEnvoyClientBuilder(getApplication())
     .addLogLevel(LogLevel.WARN)
-    .addStatsFlushSeconds(60)
     ...
     .build()
 
 **Swift example**::
 
   let envoy = try EnvoyClientBuilder()
     .addLogLevel(.warn)
-    .addStatsFlushSeconds(60)
     ...
     .build()
 
-After the client is created, it should be stored and kept in memory in order to be used
-for issuing requests.
+----------------------
+``EnvoyClientBuilder``
+----------------------
+
+This type is used to configure an instance of ``EnvoyClient`` before finally
+creating the client using ``.build()``.
+
+Available builders are 1:1 between iOS/Android, and are documented below.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``addConnectTimeoutSeconds``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Specify the timeout for new network connections to hosts in Envoy Mobile clusters.
+
+**Example**::
+
+  // Kotlin
+  builder.addConnectTimeoutSeconds(30L)
+
+  // Swift
+  builder.addConnectTimeoutSeconds(30)
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``addDNSFailureRefreshSeconds``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Specify the rate at which Envoy Mobile should refresh DNS in states of failure.
+
+This should typically be a relatively aggressive range compared to the standard-state DNS refresh
+rate, as it is required for Envoy Mobile to recover and continue making requests.
+
+**Example**::
+
+  // Kotlin
+  builder.addDNSFailureRefreshSeconds(2, 5)
+
+  // Swift
+  builder.addDNSFailureRefreshSeconds(base: 2, max: 5)
+
+~~~~~~~~~~~~~~~~~~~~~~~~
+``addDNSRefreshSeconds``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Specify the interval at which Envoy should forcefully refresh DNS.
+
+**Example**::
+
+  // Kotlin
+  builder.addDNSRefreshSeconds(60L)
+
+  // Swift
+  builder.addDNSRefreshSeconds(60)
+
+~~~~~~~~~~~~~~~
+``addLogLevel``
+~~~~~~~~~~~~~~~
+
+Specify the log level to be used when running the underlying Envoy engine.
+
+**Example**::
+
+  // Kotlin
+  builder.addLogLevel(LogLevel.WARN)
+
+  // Swift
+  builder.addLogLevel(.warn)
+
+~~~~~~~~~~~~~~~~~~
+``addStatsDomain``
+~~~~~~~~~~~~~~~~~~
+
+Specify a domain which implements the
+:tree:`stats endpoint <83908423d46a37574e9a35627df1f3dd9634e5ec/library/common/config_template.cc#L139-L145>`
+in order to take advantage of the
+`stats emitted by Envoy <https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats>`_
+(and subsequently Envoy Mobile).
+
+Note that only stats specified in the configuration's
+:tree:`whitelist <83908423d46a37574e9a35627df1f3dd9634e5ec/library/common/config_template.cc#L146-L167>`
+will be emitted.
+
+**Example**::
+
+  // Kotlin
+  builder.addStatsDomain("io.envoyproxy.envoy-mobile")
```

Comment:
Just a nit: it would make more sense if it was something like `https://envoyproxy.envoy-mobile.io`?

Reply:
sure